### PR TITLE
Improve auth UX

### PR DESCRIPTION
**Pull request by dpordomingo**
caused by #32

Some minor improvements for the UX:
- Afther logging out, the user is redirected to the Home.
- If there is no user logged in, it won't be requested the user data to the API

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -2,7 +2,15 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './App';
 
+import * as Auth0 from './react-auth0-spa';
+
 it('renders without crashing', () => {
+
+  jest
+    .spyOn(Auth0, 'useAuth0')
+    .mockImplementation(() => ({}))
```
Comment (dpordomingo) [inline src/App.test.js]: This commit c73780cfed3fb5b02d6f6d71eede7ac6ea7dc807 mocks `useAuth0` hook with an empty context value.

Auth0 React wrapper (`react-auth0-spa.js`) is initializing the `Auth0Context` [without a value](https://github.com/athenianco/athenian-webapp/blob/master/src/react-auth0-spa.js) because it will be assigned later on rendering.

Turns out that tests do not behave like the browser regarding the React Context and Hooks, so, in that case, it returns nothing when called from a component rendered by the test suite.

In order to avoid a null pointer exception in those situations,
```
   "TypeError: Cannot read property 'isAuthenticated' of undefined"
```
it was needed to mock the content returned by `useAuth0` in order to get some value from there

TL;DR
I got a hint to get this solution in this post about [Testing ‘useContext’ React hook](https://medium.com/7shifts-engineering-blog/testing-usecontext-react-hook-with-enzyme-shallow-da062140fc83)

File: src/pages/auth/Logout.jsx [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+  const history = useHistory();
   const { loading, isAuthenticated, logout } = useAuth0();
 
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      history.push('/');
+    }, 1000);
+    return () => clearTimeout(timer);
+  }, [history]);
+
```
Comment (se7entyse7en) [inline src/pages/auth/Logout.jsx]: This is not needed, the proper way to handle this is to pass a `returnTo` option to `logout`. Have a look here:
- https://github.com/auth0/auth0-spa-js/blob/902062723784d29b1bbeac9ba634071a51e38c4b/src/global.ts#L191
- https://auth0.com/docs/logout/guides/redirect-users-after-logout
- https://github.com/auth0/auth0-spa-js/blob/ee53bcb4ddd8c2620392195dc04096046a61e47f/static/index.html#L104

Comment (dpordomingo) [reply src/pages/auth/Logout.jsx]: Maybe I misunderstood the purpose of the `/logout` page:
Should it be the final stage when the user logs out? Or should it redirect to the user after saying _goodbye_?

possibilities:
- `onClick={() => logout({ returnTo: 'http://localhost:3000/' })}` &rarr; `/logout` won't be visited.
- `onClick={() => logout({ returnTo: 'http://localhost:3000/logout' })}` &rarr; `/logout` will be final.

Comment (se7entyse7en) [reply src/pages/auth/Logout.jsx]: `/logout` is called `onClick` of the `logout` button in the dropdown. See [here](https://github.com/athenianco/athenian-webapp/blob/master/src/components/layout/Navbar.jsx#L23).

Once called this "logic" page is called in order to call the `logout` function provided by the Auth0 SDK.

We could avoid having this "page" and add the first alternative that you mentioned directly in the navbar. The reason why I separated this, is in case this permits us to possibly extend the behaviour of logout more easily. For example, the login will need to open a new tab asking for GH App installation. So for consistency, I did the same for logout.

So the dropdown button will go redirect to `/logout` and will call `logout({returnTo: 'http://localhost:3000/'})`. This will cause the logout through Auth0 that will then redirect to the home.

Comment (dpordomingo) [reply src/pages/auth/Logout.jsx]: done
